src/ddb/ddb_commands.c

```diff
@@ -240,60 +242,74 @@ dump_ilog_entry_cb(void *cb_arg, struct ddb_ilog_entry *entry)
 int
 ddb_run_dump_ilog(struct ddb_ctx *ctx, struct dump_ilog_options *opt)
 {
-	struct dv_tree_path_builder	vtp = {ctx->dc_poh};
-	int				rc;
-	daos_handle_t			coh;
+	struct dv_tree_path_builder	 vtpb = {0};
+	struct dv_tree_path		*vtp = &vtpb.vtp_path;
+	daos_handle_t			 coh;
+	int				 rc;
 
 	if (!opt->path) {
-		ddb_error(ctx, "A VOS path to dump the ilog for is required.");
+		ddb_error(ctx, ilog_path_required_error_message);
 		return -DER_INVAL;
 	}
 
-	rc = init_path(ctx->dc_poh, opt->path, &vtp);
+	rc = init_path(ctx->dc_poh, opt->path, &vtpb);
 	if (!SUCCESS(rc))
 		return rc;
+	vtp_print(ctx, vtp, true);
 
-	/*
-	 * ilog for object or dkey ...
-	 * Should have a path to at least the object, but not including an akey
-	 */
-	if (!dv_has_cont(&vtp.vtp_path) || !dv_has_obj(&vtp.vtp_path) ||
-	    dv_has_akey(&vtp.vtp_path)) {
-		ddb_error(ctx, "Path to object or dkey is required.\n");
-		ddb_vtp_fini(&vtp);
+	if (!dv_has_cont(vtp)) {
+		ddb_error(ctx, ilog_path_required_error_message);
 		return -DER_INVAL;
 	}
 
-	vtp_print(ctx, &vtp.vtp_path, true);
-	rc = dv_cont_open(ctx->dc_poh, vtp.vtp_path.vtp_cont, &coh);
+	rc = dv_cont_open(ctx->dc_poh, vtp->vtp_cont, &coh);
 	if (!SUCCESS(rc)) {
-		ddb_vtp_fini(&vtp);
+		ddb_vtp_fini(&vtpb);
 		return rc;
 	}
-	rc = dv_get_obj_ilog_entries(coh, vtp.vtp_path.vtp_oid,
-				     dump_ilog_entry_cb, ctx);
+
+	if (dv_has_akey(vtp)) {
+		rc = dv_get_key_ilog_entries(coh, vtp->vtp_oid, &vtp->vtp_dkey, &vtp->vtp_akey,
+					     dump_ilog_entry_cb, ctx);
+	} else if (dv_has_dkey(vtp)) {
+		rc = dv_get_key_ilog_entries(coh, vtp->vtp_oid, &vtp->vtp_dkey, NULL,
+					     dump_ilog_entry_cb, ctx);
+	} else if (dv_has_obj(vtp)) {
+		rc = dv_get_obj_ilog_entries(coh, vtpb.vtp_path.vtp_oid, dump_ilog_entry_cb, ctx);
+	} else {
+		ddb_error(ctx, ilog_path_required_error_message);
+		rc = -DER_INVAL;
+	}
+
 	dv_cont_close(&coh);
-	ddb_vtp_fini(&vtp);
+	ddb_vtp_fini(&vtpb);
 
 	return rc;
 }
 
+struct dtx_cb_args {
+	struct ddb_ctx *ctx;
+	uint32_t entry_count;
+};
+
 static int
 active_dtx_cb(struct dv_dtx_active_entry *entry, void *cb_arg)
 {
-	struct ddb_ctx *ctx = cb_arg;
+	struct dtx_cb_args *args = cb_arg;
 
-	ddb_print_dtx_active(ctx, entry);
+	ddb_print_dtx_active(args->ctx, entry);
+	args->entry_count++;
 
 	return 0;
 }
 
 static int
 committed_cb(struct dv_dtx_committed_entry *entry, void *cb_arg)
```

Comment:
Better to rename it as "committed_dtx_cb" to consistency with the name "active_dtx_cb".